[UIAsyncTextInput] Respect UIKit/async_text_input_(iphone|ipad) after rdar://119688719

whsieh opened the pull request:
#### cc5c0bf1476ad4be35974c83d93b4586fc77fda7
<pre>
[UIAsyncTextInput] Respect UIKit/async_text_input_(iphone|ipad) after <a href="https://rdar.apple.com/119688719">rdar://119688719</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=266586">https://bugs.webkit.org/show_bug.cgi?id=266586</a>
<a href="https://rdar.apple.com/119818867">rdar://119818867</a>

Reviewed by Richard Robinson.

UIKit is splitting the `async_text_input` feature flag out into `async_text_input_iphone` and
`async_text_input_ipad` in <a href="https://rdar.apple.com/119688719">rdar://119688719</a>. To keep async text input enabled after this change, we
need to tweak our enablement logic to match these new names.

* Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm:
(WebKit::isAsyncTextInputFeatureFlagEnabled):

Pull this feature flag check out into a common helper method, and also continue checking against the
old `UIKit/async_text_input` name to make sure we don&apos;t inadvertently disable the feature on current
versions of UIKit in the build.

(WebKit::defaultUseAsyncUIKitInteractions):
(WebKit::defaultWriteRichTextDataWhenCopyingOrDragging):

Canonical link: <a href="https://commits.webkit.org/272242@main">https://commits.webkit.org/272242@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/537a603c797a2ed08ba94d08c8cd89906f802953

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31007 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9681 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32696 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33521 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28010 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12028 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6944 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27861 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31339 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8156 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27737 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7003 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7175 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34859 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28230 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28087 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33310 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7210 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5279 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31142 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8911 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7317 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7918 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7757 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm:
```diff
@@ -75,36 +75,33 @@ bool defaultMediaSourceEnabled()
+        if (PAL::deviceClassIsSmallScreen())
+            enabled = os_feature_enabled(UIKit, async_text_input_iphone) || os_feature_enabled(UIKit, async_text_input);
+        else if (PAL::deviceHasIPadCapability())
+            enabled = os_feature_enabled(UIKit, async_text_input_ipad);
```
Wouldn't this disable the feature on iPad for current versions of UIKit in the build?

aprotyas replied:
I was thinking the logic here would look more like this:

```cpp
        enabled = os_feature_enabled(UIKit, async_text_input);
        if (PAL::deviceClassIsSmallScreen())
            enabled |= os_feature_enabled(UIKit, async_text_input_iphone);
        else if (PAL::deviceHasIPadCapability())
            enabled |= os_feature_enabled(UIKit, async_text_input_ipad);
```

whsieh replied:
Ah, the intent is exactly to disable the feature on iPad for now (but temporarily keep support for `async_text_input`, as an alias for `async_text_input_iphone`).

(In other words, we don't want to do this:)
```
    enabled = os_feature_enabled(UIKit, async_text_input);
    if (PAL::deviceClassIsSmallScreen())
        enabled |= os_feature_enabled(UIKit, async_text_input_iphone);
    else if (PAL::deviceHasIPadCapability())
        enabled |= os_feature_enabled(UIKit, async_text_input_ipad);
```
...since it'll enable the feature on iPad.

aprotyas replied:
Oh I see! My bad. Then this is good to go.

aprotyas reviewed (commented):
LGTM with a question

whsieh committed:
[UIAsyncTextInput] Respect UIKit/async_text_input_(iphone|ipad) after rdar://119688719
https://bugs.webkit.org/show_bug.cgi?id=266586
rdar://119818867

Reviewed by Richard Robinson.

UIKit is splitting the `async_text_input` feature flag out into `async_text_input_iphone` and
`async_text_input_ipad` in rdar://119688719. To keep async text input enabled after this change, we
need to tweak our enablement logic to match these new names.

* Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm:
(WebKit::isAsyncTextInputFeatureFlagEnabled):

Pull this feature flag check out into a common helper method, and also continue checking against the
old `UIKit/async_text_input` name to make sure we don't inadvertently disable the feature on current
versions of UIKit in the build.

(WebKit::defaultUseAsyncUIKitInteractions):
(WebKit::defaultWriteRichTextDataWhenCopyingOrDragging):

Canonical link: https://commits.webkit.org/272242@main
